### DISCUSSION TITLE
Maven multi-module project handled incorrectly

### DISCUSSION THREAD

**Discussion by skorhone** [renovatebot/renovate#21742] [Suggest an Idea]
**What Renovate type are you using?**
Using self-hosted from latest container image.

**Describe the bug**
When multi-module project has dependencies between modules, renovate bot tries to manage their versions. If current snapshot version isn't the latest artifact in repository or if snapshot isn't in repository, this results to a wrong result.

Renovate should not dependencies that are contained in multi-module project. I would expect that similar behavior is wanted in large mono repositories

**Did you see anything helpful in debug logs?**
Nothing useful in logs: #185507657

**To Reproduce**
Project:

https://github.com/skorhone/renovate-maven-multimodule

Generated:

https://github.com/skorhone/renovate-maven-multimodule/pull/2

Which contains entries such as:

Update dependency org.apache.rocketmq:rocketmq-spring-all to v2.1.0

**Additional context**
-

**Comment by viceice** [comment]
you should disable renovate updates for local modules as workaround

**Comment by skorhone** [comment]
@viceice Is there a config option that disables all local modules, or do I need to specify them each in config?

I'll document suggested workaround to example project

**Comment by viceice** [comment]
you need to use a `packageRule` with [`packagePatterns`](https://docs.renovatebot.com/configuration-options/#packagepatterns)

**Comment by viceice** [comment]
```json
{
...
  "packageRules": [
    {
      "packagePatterns": [ "^org\\.apache\\.rocketmq"],
      "enabled": false
    }
  ]
...
}

**Comment by rjatkins** [comment]
Detecting a mono-versioned multi-module maven project would involve traversing all the poms in the project to inspect their project.version and project.parent.version elements. if the project.version isn't present, or the two versions are the same (typically: matching x.y.z-SNAPSHOT versions), and the parent pom present in the source tree, then the version of the child pom is managed by its parent. If all the descendant poms are versioned this way, then the whole project is mono-versioned.

**Comment by arkel-s** [comment]
Ugly workaround: make those internal modules not managed by Renovate:
```json
  "packageRules": [{
      "matchPackageNames": ["com.mycompany:my-internal-module"],
      "enabled": false
    }]
```
